packages/vite/src/node/server/index.ts

```diff
@@ -618,6 +618,11 @@ export async function _createServer(
       return ssrRewriteStacktrace(stack, server.environments.ssr.moduleGraph)
     },
     async reloadModule(module) {
+      warnFutureDeprecation(
+        config,
+        'removeServerModuleGraph',
+        'server.reloadModule is deprecated. Use environment.reloadModule instead.',
+      )
```

Comment:
Shouldn't this be given a new deprecation name like `'removeServerReloadModule'`? Otherwise the default deprecation message is a bit unrelated:
https://github.com/vitejs/vite/blob/e49f50599d852eec644e79b074b4648e2dff1e5d/packages/vite/src/node/deprecations.ts#L23-L24

Reply:
I thought the list of deprecation may be too many and this API is part of the `server.moduleGraph` API as this requires passing a module node. Maybe updating the message like https://github.com/vitejs/vite/pull/20431/files#diff-c14a5267085209d74a02db5cd5a4dd27347ae0ebedb7561f5a8babea226573a9L27 ?

Reply:
It's a mix of HMR and the module graph, so I think it doesn't quite fit still. The deprecations are indeed getting more but I think being granular is probably also good for narrowing down one-by-one for users.

So far the deprecations are all related to a specific API. I know I've approved https://github.com/vitejs/vite/pull/20435 though that makes the deprecation like an umbrella of APIs, but I figured those APIs are often used closely together so it's kinda fine.

Maybe @patak-dev has an opinion here on how to organize it?

Reply:
Yes, maybe it could be a separate deprecation name. It could also be as sapphi suggested updating the default description to be "server.moduleGraph and functions using it is deprecated (list of related methods)"?.